subsys/usb/device/usb_descriptor.c

```diff
@@ -347,12 +347,11 @@ static void usb_fix_ascii_sn_string_descriptor(struct usb_sn_descriptor *sn)
 	default_sn_len = strlen(CONFIG_USB_DEVICE_SN);
 
 	if (runtime_sn_len != default_sn_len) {
-		LOG_ERR("the new SN descriptor doesn't have the same "
+		LOG_WRN("the new SN descriptor doesn't have the same "
 			"length as CONFIG_USB_DEVICE_SN");
-		return;
 	}
 
-	memcpy(sn->bString, runtime_sn, runtime_sn_len);
+	memcpy(sn->bString, runtime_sn, MIN(runtime_sn_len, default_sn_len));
```

Comment:
What about the trailing bytes? Shouldn't bLength be reduced to reflect shorter string?

Reply:
the `sn->bString` already has the original string with zero byte there; so memcpy here always copy non-zero bytes only.
The leftover byte in `runtime_sn` will be ignored, which means that for the SN generation used not the lowest bits of entropy; but we talk about 4 bits loss here only.